tests/utilities/nidigital_measurement/__init__.py

```diff
@@ -0,0 +1,114 @@
+
+import nidigital
+
+import ni_measurementlink_service as nims
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "NIDigitalMeasurement.serviceconfig",
+    version="0.1.0.0",
+    ui_file_paths=[
+        service_directory,
+    ],
+)
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("pin_names", nims.DataType.StringArray1D, ["CS"])
+@measurement_service.configuration("multi_session", nims.DataType.Boolean, False)
+@measurement_service.output("session_names", nims.DataType.StringArray1D)
+@measurement_service.output("resource_names", nims.DataType.StringArray1D)
+@measurement_service.output("channel_lists", nims.DataType.StringArray1D)
+@measurement_service.output("connected_channels", nims.DataType.StringArray1D)
+@measurement_service.output("passing_sites", nims.DataType.Int32Array1D)
+@measurement_service.output("failing_sites", nims.DataType.Int32Array1D)
+def measure(
+    pin_names: Iterable[str],
+    multi_session: bool,
+) -> Tuple[
+    Iterable[str], Iterable[str], Iterable[str], Iterable[str], Iterable[str], Iterable[str]
+]:
+    """NI-Digital MeasurementLink test service."""
+    if multi_session:
+        with measurement_service.context.reserve_sessions(pin_names) as reservation:
+            with reservation.initialize_nidigital_sessions() as session_infos:
+                connections = reservation.get_nidigital_connections(pin_names)
+                assert all([session_info is not None for session_info in session_infos])
+                passing_sites, failing_sites = _burst_spi_pattern(session_infos)
+
+                return (
+                    [session_info.session_name for session_info in session_infos],
+                    [session_info.resource_name for session_info in session_infos],
+                    [session_info.channel_list for session_info in session_infos],
+                    [connection.channel_name for connection in connections],
+                    list(passing_sites),
+                    list(failing_sites),
+                )
+    else:
+        with measurement_service.context.reserve_session(pin_names) as reservation:
+            with reservation.initialize_nidigital_session() as session_info:
+                connection = reservation.get_nidigital_connection(list(pin_names)[0])
+                assert session_info is not None
+                passing_sites, failing_sites = _burst_spi_pattern([session_info])
+
+                return (
+                    [session_info.session_name],
+                    [session_info.resource_name],
+                    [session_info.channel_list],
+                    [connection.channel_name],
+                    passing_sites,
+                    failing_sites,
+                )
+
+
+def _burst_spi_pattern(
+    session_infos: Sequence[nims.session_management.TypedSessionInformation[nidigital.Session]],
```

Comment:
[style] `nims.session_management.TypedSessionInformation` is overly long. Use `from ... import ...` instead.